presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayRemoveNullsFunction.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.common.block.Block;
+import com.facebook.presto.common.block.BlockBuilder;
+import com.facebook.presto.common.type.Type;
+import com.facebook.presto.spi.function.Description;
+import com.facebook.presto.spi.function.ScalarFunction;
+import com.facebook.presto.spi.function.SqlType;
+import com.facebook.presto.spi.function.TypeParameter;
+
+import java.util.HashSet;
+
+import static com.facebook.presto.common.type.TypeUtils.readNativeValue;
+
+@ScalarFunction("remove_nulls")
+@Description("Removes null values from an array")
+public final class ArrayRemoveNullsFunction
+{
+    @TypeParameter("E")
+    public ArrayRemoveNullsFunction(@TypeParameter("E") Type elementType) {}
+
+    @TypeParameter("E")
+    @SqlType("array(E)")
+    public Block remove(
+            @TypeParameter("E") Type type,
+            @SqlType("array(E)") Block array)
+    {
+        HashSet<Integer> indicesToSkip = new HashSet<>();
+        for (int i = 0; i < array.getPositionCount(); i++) {
+            Object element = readNativeValue(type, array, i);
+            if (element == null) {
+                indicesToSkip.add(i);
+            }
+        }
+
+        if (indicesToSkip.isEmpty()) {
+            // all elements are non-null
+            return array;
+        }
```

Comment:
Nice optimization! Even better- a) use block.mayHaveNulls to make it even faster and b) (optional) also when you find the first null at k break, copy 1->k-1 into block builder then build the rest with null check inline

Reply:
done!